client/src/components/GraphViewerComponent/GraphViewerRelationshipCreateComponent/GraphViewerRelationshipCreateComponent.js

```diff
@@ -93,56 +106,98 @@ export class GraphViewerRelationshipCreateComponent extends Component {
+    this.setState({ isLoading: true });
+    setTimeout(() => {
+      document.getElementById("create-relationship-heading").focus();
+    }, 200);
+    this.setState(({ hasSwapExecuted }) => ({ hasSwapExecuted: !hasSwapExecuted }));
 
-    return { sourceId, targetId };
+    try {
+      const { selectedNode, selectedNodes } = this.props;
+      let source = selectedNodes.find(x => x.id !== selectedNode.id);
+      let target = selectedNode;
+      if (this.state.hasSwapExecuted) {
+        source = selectedNode;
+        target = selectedNodes.find(x => x.id !== selectedNode.id);
+      }
+      this.setState({sourceId: source.id});
+      this.setState({targetId: target.id});
+      const relationshipItems = await new ModelService().getRelationships(source.modelId, target.modelId);
+      if (relationshipItems.length === 0) {
+        this.setState({hasRelationships: false});
+      } else {
+        this.setState({hasRelationships: true});
+      }
+      this.setState({ relationshipItems });
+    } catch (exc) {
+      this.setState({ relationshipItems: [] });
+      exc.customMessage = `Error in retrieving model. Requested ${this.state.sourceModelId}`;
+      eventService.publishError(exc);
+    }
+    this.setState({ isLoading: false });
   }
 
   render() {
-    const { relationshipItems, relationshipId, isLoading, showModal, hasRequiredRelError } = this.state;
-    const { sourceId, targetId } = this.getNodes();
+    const { relationshipItems, relationshipId, isLoading, showModal, hasRequiredRelError, sourceId, targetId } = this.state;
 
     return (
       <ModalComponent isVisible={showModal} isLoading={isLoading} className="gc-dialog">
         <h2 className="heading-2" tabIndex="0" id="create-relationship-heading">Create Relationship</h2>
         <h4>Source ID</h4>
         <TextField disabled readOnly id="sourceIdField" ariaLabel="Source ID" className="modal-input" styles={this.getStyles} value={sourceId} />
+        <div className="btn-icon" style={{ float: "right" }}>
+          <IconButton iconOnly="true" iconProps={this.swapRelationshipIcon} title="Swap Relationship" ariaLabel="Swap Relationship" onClick={this.swap} />
+        </div>
         <h4>Target ID</h4>
         <TextField disabled readOnly id="targetIdField" ariaLabel="Target ID" className="modal-input" styles={this.getStyles} value={targetId} />
         <h4>Relationship</h4>
-        <Dropdown
-          tabIndex="0"
-          ariaLabel="Select an option"
-          required
-          placeholder="Select an option"
-          className="modal-input"
-          selectedKey={relationshipId}
-          options={relationshipItems.map((q, i) => ({ key: i, text: q }))}
-          styles={{
-            dropdown: { width: 208 }
-          }}
-          errorMessage={hasRequiredRelError ? "Please select a relationship" : null}
-          onChange={this.onSelectedRelChange} />
+        <div style={{ display: this.state.hasRelationships ? "block" : "none" }}>
```

Comment:
Rather than altering display, can we just not render the element?

E.g. `{ hasRelationships && <div ... }`

Reply:
Change Complete